### DISCUSSION TITLE
Testing failed: No such module 'ComposableArchitecture'

### DISCUSSION THREAD

**Discussion by chrisballinger** [pointfreeco/swift-composable-architecture#558] [Q&A] [answered]
We've been experiencing a lot of issues with TCA on CI where the tests compile and pass but Xcode thinks that it can't find the module, and then fails it all. Possibly related to this issue: https://github.com/pointfreeco/swift-composable-architecture/issues/22

```
	 Executed 20 tests, with 0 failures (0 unexpected) in 6.760 (6.780) seconds

Testing failed:
	No such module 'ComposableArchitecture'

** TEST FAILED **
```

The frustrating thing is that everything works locally, and this *was* only an occasional failure, but now it is happening 100% of the time (only on CI). Our CI is running Xcode 12.4 and we are running both Xcode 12.4 and Xcode 12.5 locally.

I think that this is an Xcode / Swift Package Manager bug, and the best workaround might be providing alternative library packaging (like CocoaPods): https://github.com/pointfreeco/swift-composable-architecture/issues/557

edit: workaround here https://github.com/pointfreeco/swift-composable-architecture/issues/557#issuecomment-846223323

**Comment by chrisballinger** [comment]
Okay actually I can reproduce it locally, but only when testing via the command line (thru a fastlane lane). Will dig a bit deeper.

**Comment by mluisbrown** [comment]
Are you using a custom build configuration name (ie, something other than `Debug`)? That was the issue in https://github.com/pointfreeco/swift-composable-architecture/issues/22#issuecomment-628170646

FWIW we've never seen this issue in our CI, both with Xcode 12.4 and 12.5.

**Comment by chrisballinger** [comment]
We are using some custom scheme names, e.g.

```ruby
# Podfile
project 'XXXX', 'Develop Debug' => :debug, 'Sprint Debug' => :debug, 'AppStore Debug' => :debug,
  'Develop Release' => :release, 'Sprint Release' => :release, 'AppStore Release' => :release
```

The target that is failing is an Xcode static library target `Services` that utilizes TCA in a number of places. However the problem is even stranger than I thought. There is a single source file that fails like this, multiple times:

```
❌  /XXXX/Services/XXXX.swift:8:8: no such module 'ComposableArchitecture'
```

printed over and over, when compiling other unrelated source files in that same target.

The top of the file looked like this:

```swift
import ComposableArchitecture
import Swiftilities
```

so I changed the order to this:

```swift
import Swiftilities
import ComposableArchitecture
```

and that may have fixed the problem.

**Comment by mbrandonw** [comment]
Hi @chrisballinger, sorry you're having troubles, but this is most likely a bug in Xcode that needs to be worked around, so I'm going to move it to a discussion rather than an issue on this repo.

**Comment by chrisballinger** [comment, marked as answer]
I can confirm that ditching SPM and using CocoaPods resolves this issue. Workaround here: https://github.com/pointfreeco/swift-composable-architecture/issues/557#issuecomment-846223323
